### PR TITLE
[CL-41] Typography directive

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: libs/components/src/typography/typography.directive.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Directive, HostBinding, Input } from "@angular/core";
+
+type TypographyTypes = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "body1" | "body2" | "helper";
+
+const styles: Record<TypographyTypes, string[]> = {
+  h1: ["tw-text-3xl", "tw-font-semibold"],
```
Comment (Hinton) [inline libs/components/src/typography/typography.directive.ts:6]: Do we want to use h1 as name, or have more semantic meaning?

**Comment by coroiu** [comment]
Ah yeah, fair point about bootstrap messing that up!

File: libs/components/src/typography/typography.directive.ts (hunk 1 above)
Comment (coroiu) [reply libs/components/src/typography/typography.directive.ts:6]: I think this looks fine!

File: libs/components/src/typography/typography.directive.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+@Directive({
+  selector: "[bitTypography]",
+})
+export class TypographyDirective {
+  @Input("bitTypography") bitTypography: TypographyTypes;
+
+  @HostBinding("class") get classList() {
+    return styles[this.bitTypography] ?? ["hi"];
```
Comment (coroiu) [inline libs/components/src/typography/typography.directive.ts:24]: question: what is `hi`?

File: libs/components/src/typography/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+// export * from "./typography.module";
```
Comment (coroiu) [inline libs/components/src/typography/index.ts:1]: comment

**Review by coroiu — COMMENTED** [review]
Started reviewing and then realized that this PR is marked as `draft` 🤷

File: libs/components/src/typography/typography.directive.ts (hunk 2 above)
Comment (Hinton) [reply libs/components/src/typography/typography.directive.ts]: A friendly, informal, casual greeting said upon someone's arrival.

**Comment by Hinton** [comment]
We need to figure out what margins to use. @danielleflinn something you can assist with perhaps?

**Comment by willmartian** [comment]
What is the benefit to a directive-based approach over CSS classes?

**Comment by Hinton** [comment]
None really. The jira ticket covers the different approaches and for our use case classes works just as well as angular directives.

If I remember correctly the thinking was that we already have directives and introducing classes as another abstraction layer could be confusing.

File: libs/components/src/stories/typography.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { Meta, Story, Source } from "@storybook/addon-docs";
+
+<Meta title="Documentation/Typography" />
+
+# Typography
+
+<Story id="component-library-typography--h-1" />
+
+```html
+<h1 bitTypotgraphy="h1">H1</h1>
```
Comment (coroiu) [inline libs/components/src/stories/typography.stories.mdx:10]: typo: 
```suggestion
<h1 bitTypography="h1">H1</h1>
```

same for the rest of the rows in this file

**Comment by coroiu** [comment]
With directives we could add a flag that e.g. disables bottom margin. CSS classes could do the same but I guess it would be easier to discover the proper way of doing it by checking the directive source code

**Review by coroiu — CHANGES_REQUESTED** [review]
Looks good, except for typo 😄

**Review by coroiu — CHANGES_REQUESTED** [review]
You missed the rest of the typos in `typography.stories.mdx`

**Review by coroiu — APPROVED** [review]

**Pull request merged**